system/modules/admin/partials/actions/listcomments.php

```diff
@@ -6,4 +6,48 @@ function listcomments(\Web $w, $params) {
     $w->ctx("comments", $w->Comment->getCommentsForTable($object->getDbTableName(), $object->id));
     $w->ctx("redirect", $redirect);
     $w->ctx("object", $object);
-}
\ No newline at end of file
+    
+    //get recipients for comment notifications
+    $get_recipients = $w->callHook('comment', 'get_notification_recipients_' . $object->getDbTableName(),['object_id'=>$object->id]);
+    //add checkboxes to the form for each notification recipient    
+    $recipients_form_html = '';
+    if (!empty($get_recipients)) {
+        $unique_recipients = [];
+        foreach($get_recipients as $recipients) {
+            foreach ($recipients as $user_id => $is_notify) {
+                if($is_notify && !array_key_exists($user_id, $unique_recipients)){
+                    $unique_recipients[$user_id] = $is_notify;
+                }
+            }
+        }
+        $recipients_form_html .= '<h4>Notifications</h4>';
+        $recipients_form_html .= '<input type="hidden" name="is_notifications" value="1" id="is_notifications">';
+        $recipients_form_html .= '<div id="notifications_list">';
+        $parts = array_chunk($unique_recipients, 4, true);
+        foreach ($parts as $key=>$row) {
+            
+            $recipients_form_html .= '<ul class="small-block-grid-1 medium-block-grid-' . count($row) . ' section-body">';
+            foreach ($row as $user_id => $is_notify) {
+                $user = $w->Auth->getUser($user_id);
+                if (!empty($user)) {
+                    $recipients_form_html .= '<li><label class="small-12 columns">';
```

Comment:
On that, try and avoid building HTML in actions. Don't forget you can still use php functions like array_chunk and foreach loops to build javascript functionality. In the interest of getting this feature in, leave it in the action for now but please tidy up string building as per above.